homeassistant/auth/auth_store.py

```diff
@@ -120,9 +120,6 @@ async def async_create_user(
 
         new_user = models.User(**kwargs)
 
-        while new_user.id in self._users:
-            new_user = models.User(**kwargs)
-
```

Comment:
Simply reverting will not fix the original issue.

See also original comment from @elupus

> If the uuid is mocked, this would (could) be a busy wait. Not sure that is better. Why not just assert and crash?

Reply:
Feel free to open an alternative PR.

Reply:
uuid.uuid4() has doesn't even have a 50% chance of collision until you get into quadrillions of generations so raising an exception should be fine here instead